app/src/main/java/org/mozilla/focus/home/pocket/PocketVideoFragment.kt

```diff
@@ -53,31 +56,38 @@ class PocketVideoFragment : Fragment() {
     }
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
-        fun displayFeed(layout: View, videos: PocketVideosDeferred) {
-            layout.videoFeed.gridView.adapter = PocketVideoAdapter(context!!, videos, fragmentManager!!)
-        }
-
-        val layout = inflater.inflate(R.layout.fragment_pocket_video, container, false)
+        return inflater.inflate(R.layout.fragment_pocket_video, container, false)
+    }
 
-        if (deferredVideos.isCompleted) {
-            displayFeed(layout, deferredVideos)
-        } else {
-            // TODO: #769?: display loading screen.
-
-            launch(UI) {
-                val videos = deferredVideos.await()
-                if (videos != null) {
-                    displayFeed(layout, deferredVideos)
-                } else {
-                    // TODO: #769: display error screen.
-                }
+    override fun onStart() {
+        super.onStart()
+        val adapter = PocketVideoAdapter(context!!, fragmentManager!!).also {
+            videoFeed.gridView.adapter = it
+        }
+        fun setupLoadingIfApplicable() {
+            if (deferredVideos.isNotCompleted()) {
+                // We show 20 placeholders because our feed will be updated to show 20 items
+                val placeholders = List(20) { PocketPlaceholder }
```

Comment:
nit: constant

Reply:
I'm not sure I agree.  Since the value is only used here, it seems like an additional level of indirection that we don't need.